public/types.ts

```diff
@@ -34,3 +33,11 @@ export interface ObservabilitySetup {}
 
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface ObservabilityStart {}
+
+/**
+ * Introduce a compile dependency on dashboards-assistant
+ * as observerability need some types from the plugin.
+ * It will gives an type error when dashboards-assistant is not installed so add a ts-ignore to suppress the error.
+ */
+// @ts-ignore
+export type { AssistantSetup } from "../../dashboards-assistant/public";
```

Comment:
also add this to https://github.com/opensearch-project/dashboards-observability/blob/1f9cc08be2398b1fe8bc6c20075674956857be05/public/types.ts#L23-L29?
`assistantDashboards?: AssistantSetup;`

Reply:
Sure, done for this.